### PR TITLE
chore: add lint scripts to package.json

### PR DESCRIPTION
**Pull request by sdstolworthy**
#### Description of changes

Adds helpful lint scripts to `package.json`. Uses `lint-check` script in CircleCI pipeline.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,8 @@
     "link-aa-win": "node ./scripts/link-bin.js packages/amplify-app/bin/amplify-app amplify-app-dev",
     "link-dev": "cd packages/amplify-cli && ln -s \"$(pwd)/bin/amplify\" \"$(yarn global bin)/amplify-dev\" && cd -",
     "link-win": "node ./scripts/link-bin.js packages/amplify-cli/bin/amplify amplify-dev",
+    "lint": "yarn eslint . --ext .js,.jsx,.ts,.tsx",
+    "lint-check": "yarn eslint . --ext .js,.jsx,.ts,.tsx --max-warnings=1085",
```
Comment (sobolk) [inline package.json:28]: Do we need `lint` ?

File: .circleci/config.base.yml [hunk 2]
```diff
@@ -162,7 +162,7 @@ jobs:
           key: amplify-cli-repo-{{ .Branch }}-{{ .Revision }}
       - restore_cache:
           key: amplify-cli-yarn-deps-{{ .Branch }}-{{ checksum "yarn.lock" }}
-      - run: yarn eslint . --ext .js,.jsx,.ts,.tsx --max-warnings 1085
+      - run: yarn lint-check
```
Comment (sobolk) [inline .circleci/config.base.yml:165]: could you also create aliases for linting package json and prettier check ?

File: package.json (hunk 1 above)
Comment (sobolk) [reply package.json:28]: for local development we might want to add
`yarn check-static-analysis` or something like this which will run lint, lint package json and prettier as one command.

File: package.json [hunk 3]
```diff
@@ -24,12 +24,15 @@
     "link-aa-win": "node ./scripts/link-bin.js packages/amplify-app/bin/amplify-app amplify-app-dev",
     "link-dev": "cd packages/amplify-cli && ln -s \"$(pwd)/bin/amplify\" \"$(yarn global bin)/amplify-dev\" && cd -",
     "link-win": "node ./scripts/link-bin.js packages/amplify-cli/bin/amplify amplify-dev",
+    "lint-check": "yarn eslint . --ext .js,.jsx,.ts,.tsx --max-warnings=1085",
     "lint-fix": "git diff --name-only --cached --diff-filter d | grep -E '\\.(js|jsx|ts|tsx)$' | xargs eslint --fix --quiet",
+    "lint-package-json": "yarn eslint --no-eslintrc --config .eslint.package.json.js '**/package.json'",
```
Comment (sobolk) [inline package.json:29]:
```suggestion
    "lint-check-package-json": "yarn eslint --no-eslintrc --config .eslint.package.json.js '**/package.json'",
```

**Review by sobolk — DISMISSED** [review]

File: package.json [hunk 4]
```diff
@@ -24,12 +24,15 @@
...
     "mergewords": "yarn ts-node ./scripts/handle-dict-conflicts.ts",
     "pkg-all-local": "yarn verdaccio-start && yarn verdaccio-connect && yarn publish-to-verdaccio && yarn pkg-all && yarn verdaccio-disconnect",
     "pkg-all": "source .circleci/local_publish_helpers.sh && generatePkgCli",
     "pkg-clean": "rimraf build out pkg/node_modules pkg/yarn.lock",
-    "prettify:changes": "git diff --name-only --diff-filter MRA dev | xargs yarn prettier --write",
+    "prettier-check": "yarn prettier --check .",
+    "prettify-changes": "git diff --name-only --diff-filter MRA dev | xargs yarn prettier --write",
```
Comment (sobolk) [inline package.json:35]:
```suggestion
    "prettify-check": "yarn prettier --check .",
    "prettify-changes": "git diff --name-only --diff-filter MRA dev | xargs yarn prettier --write",
```

Comment (sobolk) [reply package.json:35]: Or the other way round i.e. prettify->prettier

**Review by sobolk — APPROVED** [review]

**Review by kuhlmanp — APPROVED** [review]

**Pull request merged**